### PR TITLE
MAGE-18 Connection and configuration drawer for nodes with redux

### PR DESCRIPTION
**Pull request by Saifu0**
- Added a drawer on `EditorCanvas` component which will pop-up on double click of Components
- Added a new `EditorComponentDrawer` module which includes all the components of Drawer.
- Implemented `ComponentConnection` component and all the UI required component. 
- Implemented `ComponentConfiguration` component and all the UI required component. 
- Integrated redux with fake db and mocks data

### CODE REVIEW

File: src/__mocks__/nodes.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          },
+          {
+            id: uuid(),
+            name: 'something',
+            type: 'number'
+          }
+        ]
+      }
+    ],
+    configurations: {
```
Comment (parijatpurohit) [inline src/__mocks__/nodes.js:56]: This file contains schema for all the components. We shouldn't be hardcoding this config here.
Instead, we must allow every component to define these configurations just like we have defined connections above.
Something like:
```
configurations: [
{name: 'username', default: 'root', type: 'string'},
{name: 'password', default: 'root', type: 'string'},
{name: 'host', default: 'localhost', type: 'string'},
{name: 'port', default: '3000', type: 'string'},
]
```

And the actual values provided by the user must then be persisted in redux only.

File: src/components/editor/EditorComponentDrawer/AddMethodFieldModal.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    setType(event.target.value);
+  };
+
+  const handleSave = () => {
+    const data = {
+      name,
+      type
+    };
+    if (mode === 'request') {
+      props.addMethodRequestField(nodeId, connectionId, data);
```
Comment (parijatpurohit) [inline src/components/editor/EditorComponentDrawer/AddMethodFieldModal.js:63]: should extract everything from props in a separate variable.

`const {addMethodRequestField, addMethodResponseField} = props;`

File: src/components/editor/modules/redux/reducers/node.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  node: {
+    id: null,
+    connections: [],
+    configurations: []
+  },
+  loading: false,
+  error: null
+};
+
+// const updateMethodName = (state, action) => {
```
Comment (parijatpurohit) [inline src/components/editor/modules/redux/reducers/node.js:14]: is this comment needed now?

File: src/components/editor/modules/redux/actions/node.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  addMethodRequestFieldToDB,
+  updateMethodRequestFieldToDB,
+  removeMethodRequestFieldFromDB,
+  addMethodResponseFieldToDB,
+  updateMethodResponseFieldToDB,
+  removeMethodResponseFieldFromDB,
+  updateConfigurationToDB
+} from './db';
+
+export const getNodes = () => (dispatch) => {
```
Comment (parijatpurohit) [inline src/components/editor/modules/redux/actions/node.js:17]: are you using these actions anywhere?

File: src/components/editor/modules/redux/store.js [hunk 5]
```diff
@@ -1,6 +1,13 @@
-import { createStore } from 'redux';
-import reducer from './reducer';
+import { createStore, applyMiddleware, combineReducers } from 'redux';
+import thunk from 'redux-thunk';
```
Comment (parijatpurohit) [inline src/components/editor/modules/redux/store.js:2]: don't see the usage of thunk actions anywhere. you have created method actions above but how are we using them?

File: src/routes.js [hunk 6]
```diff
@@ -1,14 +1,14 @@
 import { Navigate } from 'react-router-dom';
 import DashboardLayout from 'src/components/DashboardLayout';
 import MainLayout from 'src/components/MainLayout';
-// import Account from 'src/pages/Account';
```
Comment (parijatpurohit) [inline src/routes.js:4]: can actually just remove these. these are mock pages that came along with the theme UI.

File: src/__mocks__/nodes.js (hunk 1 above)
Comment (Saifu0) [reply src/__mocks__/nodes.js:56]: so, we want a `Add Configuration` button which will add new configurations?

File: src/components/editor/modules/redux/reducers/node.js (hunk 3 above)
Comment (Saifu0) [reply src/components/editor/modules/redux/reducers/node.js:14]: no need. Forgot to remove.

File: src/components/editor/modules/redux/actions/node.js (hunk 4 above)
Comment (Saifu0) [reply src/components/editor/modules/redux/actions/node.js:17]: Yes, to mock the API calls. I can move this into `reducers/node.js` file if we want everything in-memory.

File: src/components/editor/EditorCanvas.js [hunk 7]
```diff
@@ -28,7 +28,7 @@ const renderMap = {
   LookAsideCache
 };
 
-const mapStateToProps = (state) => (state.canvas);
+const mapStateToProps = (state) => (state.addNode.canvas);
```
Comment (parijatpurohit) [inline src/components/editor/EditorCanvas.js:31]: check for drag and drop thing

**Review by parijatpurohit — APPROVED** [review]

**Pull request merged**